modules/RoomTree.js

```diff
@@ -1,103 +1,120 @@
-class RoomNode {
-    left = null;
-    middle = null;
-    right = null;
+import BuildDungeon from "./DungeonRoom.js";
+
+const direction = {
+    up: 0,
+    down: 1,
+    left: 2,
+    right: 3
```

Comment:
These directions could potentially be changed to match my bitflags
`const NORTH = 1;
const SOUTH = 2;
const EAST = 4;
const WEST = 8;`
But it's not important